src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -3387,16 +3400,33 @@ def _build_arguments_schema(cls, *args, **kwargs):
             is_preview=True,
             nullable=True,)
         args_schema.propagated_route_tables.Element = AAZStrArg()
-        args_schema.circuit_name = AAZStrArg(
+        args_schema.circuit_name = AAZResourceIdArg(
             options=['--circuit-name'],
             arg_group="Peering",
-            help="ExpressRoute circuit name.")
+            help="ExpressRoute circuit name.",
+            fmt=AAZResourceIdArgFormat(
+                template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Network/expressRouteCircuits/{}/peerings/"
+            )
+        )
         args_schema.associated_id._registered = False
         args_schema.propagated_ids._registered = False
         return args_schema
 
     def pre_operations(self):
         args = self.ctx.args
+        if has_value(args.peering) and '/' not in args.peering.to_serialized_data():
+            usage_error = CLIError('usage error: --peering ID | --peering NAME --circuit-name CIRCUIT')
```

Comment:
Please refine it by https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md.